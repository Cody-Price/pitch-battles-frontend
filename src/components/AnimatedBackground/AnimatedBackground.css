.animated-background {
  position: absolute;
  animation: background-move 80s linear infinite;
  width: 2100px;
  height: 1600px;
  background-repeat: no-repeat;
  background-size: cover;
  image-rendering: pixelated;
}

.landing-char1 {
  position: absolute;
  height: 144px;
  width: 78px;
  top: 350px;
  animation: landing-char1-frames 1s steps(4) infinite,
    landing-char1-move 60s linear infinite;
  z-index: 11;
  background-size: 416px 144px;
  image-rendering: pixelated;
}

.snow-backer {
  animation: snow 80s infinite linear;
  -webkit-animation: snow 80s infinite linear;
  -moz-animation: snow 80s infinite linear;
  -o-animation: snow 80s infinite linear;

  position: absolute;
  height: 6000px;
  width: 10000px;
  background: url(../Landing/images/snow/wsheet_snow_1.png);
  background-size: 384px 96px;
  z-index: 10000;
}

@keyframes landing-char1-frames {
  100% {
    background-position: -416px;
  }
}

.landing-char2 {
  position: absolute;
  height: 144px;
  width: 78px;
  top: 320px;
  animation: landing-char2-frames 1s steps(4) infinite,
    landing-char2-move 60s linear infinite;
  z-index: 10;
  background-size: 416px 144px;
  image-rendering: pixelated;
}

@keyframes landing-char2-frames {
  100% {
    background-position: -416px;
  }
}

.landing-birb {
  position: absolute;
  width: 164px;
  height: 144px;
  animation: birb-flap 1s steps(4) infinite, birb-move 20s linear infinite;
  z-index: 100;
  left: 700px;
  background-size: 656px, 144px;
}

.standing-char1 {
  position: absolute;
  z-index: 11;
  height: 123px;
  width: 62px;
  top: 580px;
  left: 605px;
  image-rendering: pixelated;
  background: url(./images/landing-standing-chars/stand.png);
  background-size: cover;
  /* animation: laughing-landing-char linear infinite 12s; */
}

.standing-char2 {
  position: absolute;
  z-index: 11;
  height: 124px;
  width: 63px;
  top: 580px;
  left: 500px;
  background: url(./images/landing-standing-chars/char2.png);
  background-size: cover;
  image-rendering: pixelated;
}

.crystal1,
.crystal2,
.crystal3,
.crystal4,
.crystal5,
.crystal6,
.crystal7,
.crystal8,
.crystal9,
.crystal10,
.crystal11,
.crystal12 {
  position: absolute;
  z-index: 18;
  height: 186px;
  width: 60px;
  top: 1000px;
  left: 1660px;
  animation: crystals 80s infinite linear;
}

.crystal2,
.crystal4,
.crystal6,
.crystal8,
.crystal10,
.crystal12 {
  left: 1080px;
}

.crystal3,
.crystal4 {
  top: 1150px;
  z-index: 19;
}

.crystal5,
.crystal6 {
  top: 1300px;
  z-index: 20;
}

.crystal7,
.crystal8 {
  top: 850px;
  z-index: 17;
}

.crystal9,
.crystal10 {
  top: 700px;
  z-index: 16;
}

.crystal11,
.crystal12 {
  top: 550px;
  z-index: 15;
}

@keyframes landing-char1-move {
  0% {
    transform: translateX(2700px);
  }
  100% {
    transform: translateX(-100px);
  }
}

@keyframes landing-char2-move {
  0% {
    transform: translateX(-100px);
  }
  100% {
    transform: translateX(2700px);
  }
}

@keyframes landing-char1 {
}

@keyframes landing-char2 {
  0% {
    background: url(./images/landing-char2/right1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  25% {
    background: url(./images/landing-char2/right1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  25.01% {
    background: url(./images/landing-char2/right2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  50% {
    background: url(./images/landing-char2/right2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  50.01% {
    background: url(./images/landing-char2/right1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  75% {
    background: url(./images/landing-char2/right1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  75.01% {
    background: url(./images/landing-char2/right3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  100% {
    background: url(./images/landing-char2/right3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
}

@keyframes background-move {
  0% {
    transform: translate(0px, -100px);

    background: url(../Game/images/snow-bg.png);
    background-size: cover;
    image-rendering: pixelated;
    background-repeat: no-repeat;
  }
  33% {
    background: url(../Game/images/snow-bg.png);
    background-size: cover;
    image-rendering: pixelated;
    background-repeat: no-repeat;
  }
  25% {
    transform: translate(-150px, -600px);
    image-rendering: pixelated;
  }
  35% {
    background: url(../Game/images/forestpath-bg.png);
    background-size: cover;
    image-rendering: pixelated;
    background-repeat: no-repeat;
  }
  50% {
    transform: translate(0px, -100px);
    image-rendering: pixelated;
  }
  64% {
    background: url(../Game/images/forestpath-bg.png);
    background-size: cover;
    image-rendering: pixelated;
    background-repeat: no-repeat;
  }
  66% {
    background: url(../Game/images/dd-bg.png);
    background-size: cover;
    image-rendering: pixelated;
    background-repeat: no-repeat;
  }
  75% {
    transform: translate(-150px, -600px);
    image-rendering: pixelated;
  }
  96% {
    background: url(../Game/images/dd-bg.png);
    background-size: cover;
    image-rendering: pixelated;
    background-repeat: no-repeat;
  }
  100% {
    background: url(../Game/images/snow-bg.png);
    background-size: cover;
    image-rendering: pixelated;
    transform: translate(-100px, -100px);
    background-repeat: no-repeat;
  }
}

@keyframes birb-flap {
  100% {
    background-position: -656px;
  }
}

@keyframes birb-move {
  0% {
    transform: translateY(1700px);
  }
  100% {
    transform: translateY(-100px);
  }
}

@keyframes crystals {
  0% {
    opacity: 0;
  }
  65% {
    opacity: 0;
  }
  66% {
    opacity: 1;
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  67% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  67.01% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  68% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  68.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  69% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  69.01% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  70% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  70.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  71% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  71.01% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  72% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  72.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  73% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  73.01% {
    opacity: 1;
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  74% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  74.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  75% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  75.01% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  76% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  76.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  77% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  77.01% {
    opacity: 1;
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  78% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  78.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  79% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  79.01% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  80% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  80.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  81% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  81.01% {
    opacity: 1;
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  82% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  82.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  83% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  83.01% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  84% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  84.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  85% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  85.01% {
    opacity: 1;
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  86% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  86.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  87% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  87.01% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  88% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  88.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  89% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  89.01% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  90% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  90.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  91% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  91.01% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  92% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  92.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  93% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  93.01% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  94% {
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  94.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  95% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  95.01% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  96% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  96.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  97% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  97.01% {
    opacity: 1;
    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  98% {
    opacity: 1;

    background: url(../Game/images/crystal1/cry3.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  98.01% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  99% {
    background: url(../Game/images/crystal1/cry2.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  99.01% {
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
  100% {
    opacity: 0;
    background: url(../Game/images/crystal1/cry1.png);
    image-rendering: pixelated;
    background-size: cover;
  }
}

@media (max-width: 700px) {
  .animated-background {
    animation: none;
    background: url(../Game/images/forestpath-bg.png);
    background-size: cover;
    image-rendering: pixelated;
    background-repeat: no-repeat;
    background-size: cover;
    image-rendering: pixelated;
  }

  .snow-backer {
    animation: none;
    background: none;
    display: none;
  }

  .landing-birb {
    left: 100px;
    z-index: 80;
  }

  .crystal1,
  .crystal2,
  .crystal3,
  .crystal4,
  .crystal5,
  .crystal6,
  .crystal7,
  .crystal8,
  .crystal9,
  .crystal10,
  .crystal11,
  .crystal12 {
    display: none;
    animation: none;
  }

  .landing-char1 {
    animation: landing-char1-frames 1s steps(4) infinite,
      landing-char1-move 25s linear infinite;
  }

  .landing-char2 {
    animation: landing-char2-frames 1s steps(4) infinite,
      landing-char2-move 20s linear infinite;
  }

  @keyframes landing-char1-move {
    0% {
      transform: translateX(800px);
    }
    100% {
      transform: translateX(-100px);
    }
  }

  @keyframes landing-char2-move {
    0% {
      transform: translateX(-100px);
    }
    100% {
      transform: translateX(800px);
    }
  }
}
