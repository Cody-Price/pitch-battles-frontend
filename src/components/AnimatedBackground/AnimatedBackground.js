import React from "react";
import PropTypes from "prop-types";
import "./AnimatedBackground.css";

const AnimatedBackground = props => {
  return (
    <div>
      <div className="animated-background">
        <div className="snow-backer" />
        <div className="landing-char1" />
        <div className="landing-char2" />
        <div className="landing-birb" />
        <div className="standing-char1" />
        <div className="standing-char2" />
        <div className="crystal1" />
        <div className="crystal2" />
        <div className="crystal3" />
        <div className="crystal4" />
        <div className="crystal5" />
        <div className="crystal6" />
        <div className="crystal7" />
        <div className="crystal8" />
        <div className="crystal9" />
        <div className="crystal10" />
        <div className="crystal11" />
        <div className="crystal12" />
      </div>
    </div>
  );
};

export default AnimatedBackground;

AnimatedBackground.propTypes = {
  instance: PropTypes.string
};
