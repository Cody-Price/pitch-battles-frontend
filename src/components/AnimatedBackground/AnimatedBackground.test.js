import React from "react";
import ReactDOM from "react-dom";
import AnimatedBackground from "./AnimatedBackground";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("AnimatedBackground", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AnimatedBackground />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
