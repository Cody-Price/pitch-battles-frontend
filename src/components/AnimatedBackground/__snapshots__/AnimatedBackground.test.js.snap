// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedBackground should match the snapshot 1`] = `
<div>
  <div
    className="animated-background"
  >
    <div
      className="snow-backer"
    />
    <div
      className="landing-char1"
    />
    <div
      className="landing-char2"
    />
    <div
      className="landing-birb"
    />
    <div
      className="standing-char1"
    />
    <div
      className="standing-char2"
    />
    <div
      className="crystal1"
    />
    <div
      className="crystal2"
    />
    <div
      className="crystal3"
    />
    <div
      className="crystal4"
    />
    <div
      className="crystal5"
    />
    <div
      className="crystal6"
    />
    <div
      className="crystal7"
    />
    <div
      className="crystal8"
    />
    <div
      className="crystal9"
    />
    <div
      className="crystal10"
    />
    <div
      className="crystal11"
    />
    <div
      className="crystal12"
    />
  </div>
</div>
`;
